mne/viz/_3d.py

```diff
@@ -957,6 +1046,185 @@ def time_label(t):
+                Flag to specify type of limits. 'value' or 'percent'.
+            ``lims`` : list | np.ndarray | tuple of float, 3 elements
+                Note: Only use this if 'colormap' is not 'mne'.
+                Left, middle, and right bound for colormap.
+            ``pos_lims`` : list | np.ndarray | tuple of float, 3 elements
+                Note: Only use this if 'colormap' is 'mne'.
+                Left, middle, and right bound for colormap. Positive values
+                will be mirrored directly across zero during colormap
+                construction to obtain negative control points.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    fig : instance of mlab.Figure
+        The mayavi figure.
+    """
+    from mayavi import mlab
+
+    fig = mlab.figure(size=fig_size, bgcolor=bgcolor, fgcolor=(0, 0, 0))
+
+    t = np.searchsorted(stc.times, time)
+    magnitude = stc.magnitude()
+
+    scale_pts, colormap, transparent = _limits_to_control_points(
+        clim, stc.data, cmap, transparent)
+
+    # The colomap can be specified both as a LUT table and a string
+    if isinstance(colormap, np.ndarray):
+        lut_table = colormap
+        cmap = 'hot'
+    else:
+        lut_table = None
+        cmap = colormap
+
+    if scale_factor is None:
+        scale_factor = 1. / (80 * np.max(magnitude.data[:, t]))
+
+    def _plot_hemi_mesh(src, hemi_ind):
+        """Plot the mesh of a single hemisphere"""
+        with warnings.catch_warnings(record=True):  # FutureWarning in traits
+            if high_resolution:
+                tris = src[hemi_ind]['tris']
+            else:
+                tris = src[hemi_ind]['use_tris']
+
+            # Plot mesh
+            mlab.triangular_mesh(src[hemi_ind]['rr'][:, 0],
+                                 src[hemi_ind]['rr'][:, 1],
+                                 src[hemi_ind]['rr'][:, 2],
+                                 tris,
+                                 color=brain_color,
+                                 opacity=opacity,
+                                 representation='surface')
+
+    def _plot_hemi_glyphs(src, hemi_ind, lut_table):
+        """Plot the glyphs of a single hemisphere"""
+        if hemi_ind == 0:
+            rr = src[hemi_ind]['rr'][stc.lh_vertno]
+            data = stc.lh_data
+            scalars = magnitude.lh_data[:, t]
+        else:
+            rr = src[hemi_ind]['rr'][stc.rh_vertno]
+            data = stc.rh_data
+            scalars = magnitude.rh_data[:, t]
+
+        q = mlab.quiver3d(
+            rr[:, 0],
+            rr[:, 1],
+            rr[:, 2],
+            data[:, 0, t],
+            data[:, 1, t],
+            data[:, 2, t],
+            scale_factor=scale_factor,
+            scalars=scalars,
+            colormap=cmap,
+            mode=mode,
+            vmin=scale_pts[0],
+            vmax=scale_pts[2],
+        )
```

Comment:
... this and maybe the scaling stuff would be the only thing you'd need to add on top of what PySurfer already gives us